elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/dynamic/DynamicAccessorValidator.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.converters.avro.dynamic
+
+import com.spotify.elitzur.MetricsReporter
+import com.spotify.elitzur.converters.avro.dynamic.dsl.AvroAccessorException._
+import com.spotify.elitzur.converters.avro.dynamic.dsl.AvroObjMapper
+import com.spotify.elitzur.validators.{
+  BaseCompanion,
+  DynamicRecordValidator,
+  Unvalidated,
+  Validator
+}
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+
+import scala.util.{Failure, Try}
+
+class DynamicAccessorValidator(
+  recordAccessorWithValidator: Array[(String, BaseCompanion[_, _], Validator[Any])], schema: Schema
```

Comment:
IIRC in the internal repo we are using the `SimpleCompanionImplicit` to fetch these, so could theoretically implicitly find that at a later stage.

Another alternative is to create a wrapper time to use instead of the Tuple3

Reply:
hmm I tried working with `SimpleCompanionImplicit` to fetch the BaseCompanion, but ran into issues since `SimpleCompanionImplicit` requires two generic types (i.e., `SimpleCompanionImplicit[T, A <: com.spotify.elitzur.validators.BaseValidationType[T]]`) and my attempts to retrieve T from  BaseValidationType[T] aren't panning out in a way that the implicits still works (i.e., with type definition `LT <: BaseValidationType[T]: ({type L[x] = SimpleCompanionImplicit[T, x]})#L,T`)

Reply:
I'm inclined to go with the alternative route and wrap it instead of the Tuple3

Reply:
Updated the PR to wrap the input instead of Tuple3